error: 'isfinite' was not declared in this scope; did you mean 'std::isfinite'?

```
cum_median.cpp: In function 'int cum_median(int, double*, double*, double*)':
cum_median.cpp:10:9: error: 'isfinite' was not declared in this scope; did you mean 'std::isfinite'?
   10 |     if(!isfinite(data_value)){
      |         ^~~~~~~~
      |         std::isfinite
In file included from /opt/local/include/LegacySupport/cmath:54,
                 from PiecewiseFunction.h:16,
                 from cum_median.cpp:3:
/opt/local/include/gcc12/c++/cmath:578:5: note: 'std::isfinite' declared here
  578 |     isfinite(_Tp __x)
      |     ^~~~~~~~
make: *** [cum_median.o] Error 1
```

```
** using staged installation
** libs
using C++ compiler: ‘g++-mp-12 (MacPorts gcc12 12.2.0_0) 12.2.0’
Warning in system2("xcrun", "--show-sdk-path", TRUE, TRUE) :
  running command ''xcrun' --show-sdk-path 2>&1' had status 64
using SDK: ‘NA’‘NA’‘NA’‘NA’‘NA’‘NA’
/opt/local/bin/g++-mp-12 -std=gnu++17 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/Rcpp/include' -isystem/opt/local/include/LegacySupport -I/opt/local/include    -fPIC  -pipe -Os -arch ppc  -c PiecewiseFunction.cpp -o PiecewiseFunction.o
/opt/local/bin/g++-mp-12 -std=gnu++17 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/Rcpp/include' -isystem/opt/local/include/LegacySupport -I/opt/local/include    -fPIC  -pipe -Os -arch ppc  -c RcppExports.cpp -o RcppExports.o
/opt/local/bin/g++-mp-12 -std=gnu++17 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/Rcpp/include' -isystem/opt/local/include/LegacySupport -I/opt/local/include    -fPIC  -pipe -Os -arch ppc  -c binseg.cpp -o binseg.o
/opt/local/bin/g++-mp-12 -std=gnu++17 -I"/opt/local/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/opt/local/Library/Frameworks/R.framework/Versions/4.3/Resources/library/Rcpp/include' -isystem/opt/local/include/LegacySupport -I/opt/local/include    -fPIC  -pipe -Os -arch ppc  -c cum_median.cpp -o cum_median.o
cum_median.cpp: In function 'int cum_median(int, double*, double*, double*)':
cum_median.cpp:10:9: error: 'isfinite' was not declared in this scope; did you mean 'std::isfinite'?
   10 |     if(!isfinite(data_value)){
      |         ^~~~~~~~
      |         std::isfinite
In file included from /opt/local/include/LegacySupport/cmath:54,
                 from PiecewiseFunction.h:16,
                 from cum_median.cpp:3:
/opt/local/include/gcc12/c++/cmath:578:5: note: 'std::isfinite' declared here
  578 |     isfinite(_Tp __x)
      |     ^~~~~~~~
make: *** [cum_median.o] Error 1
ERROR: compilation failed for package ‘binsegRcpp’
```
P. S. I do not recall what exactly was the reason for this kind of errors (we had then with a few other ports besides `binsegRcpp ` and `openexr`), it is header-related, but conclusion was that switching to `std::` fixes everything.

--- a/src/cum_median.cpp
+++ b/src/cum_median.cpp
@@ -7,7 +7,7 @@ int cum_median
   PiecewiseFunction function;
   for(int data_i=0; data_i < n_data; data_i++){
     double data_value = data_vec[data_i];
-    if(!isfinite(data_value)){
+    if(!std::isfinite(data_value)){
       return ERROR_CUM_MEDIAN_DATA_NOT_FINITE;
     }
     double weight_value = weight_vec[data_i];

